pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go

```diff
@@ -10,15 +12,16 @@ import (
 	core_xds "github.com/kumahq/kuma/pkg/core/xds"
 	"github.com/kumahq/kuma/pkg/plugins/policies/core/matchers"
 	core_rules "github.com/kumahq/kuma/pkg/plugins/policies/core/rules"
+	policies_xds "github.com/kumahq/kuma/pkg/plugins/policies/core/xds"
 	api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1"
 	v3 "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/xds"
 	xds_context "github.com/kumahq/kuma/pkg/xds/context"
 	"github.com/kumahq/kuma/pkg/xds/generator"
 )
 
 var (
-	_   core_plugins.PolicyPlugin = &plugin{}
-	log                           = core.Log.WithName("MeshTrafficPermission")
+	_   core_plugins.EgressPolicyPlugin = &plugin{}
```

Comment:
Not really part of this PR but why can't this be part of `PolicyPlugin` and just have `EgressMatchedPolicies` return an empty set in the usual case? This makes it seem like it only works on egress?

Reply:
> why can't this be part of PolicyPlugin and just have EgressMatchedPolicies return an empty set in the usual case?
Main reason was not breaking the existing API with an extra method.

Reply:
> This makes it seem like it only works on egress?

Every EgressPolicyPlugins is a PolicyPlugin:
```go
type EgressPolicyPlugin interface {
	PolicyPlugin
	// EgressMatchedPolicies returns all the policies of the plugins' type matching the external service that
	// should be applied on the zone egress.
	EgressMatchedPolicies(*core_mesh.ExternalServiceResource, xds_context.Resources) (core_xds.TypedMatchingPolicies, error)
}
```
We don't plan to have policies that work solely on Egress. 

> why can't this be part of PolicyPlugin and just have EgressMatchedPolicies return an empty set in the usual case?

Interface shows what plugin can and can't do, you don't have to go to implementation and check if it returns empty set.

Reply:
> Main reason was not breaking the existing API with an extra method.

OK yeah that makes sense for introducing it in the first place I suppose.

> We don't plan to have policies that work solely on Egress.

Yes, I know that's my point. I'm saying the name "EgressPolicy" implies to me that it works only on egress.

> Interface shows what plugin can and can't do, you don't have to go to implementation and check if it returns empty set.

I dunno, for who exactly is this useful? And either way you actually _do_ have to go to the implementation because that's the only way to see if a plugin actually implements `EgressMatchedPolicies`. Until we rewrite in Idris and have dependent types you're going to have to check the implementation to see what the plugin can and can't do.

But obviously if people feel strongly about this it doesn't matter at all.